test/e2e/network/proxy.go

```diff
@@ -133,52 +133,95 @@ var _ = common.SIGDescribe("Proxy", func() {
 			}, metav1.CreateOptions{})
 			framework.ExpectNoError(err)
 
-			// Make an RC with a single pod. The 'porter' image is
+			// Make a deployment with a single pod. The 'porter' image is
 			// a simple server which serves the values of the
 			// environmental variables below.
 			ginkgo.By("starting an echo server on multiple ports")
-			pods := []*v1.Pod{}
-			cfg := testutils.RCConfig{
-				Client:       f.ClientSet,
-				Image:        imageutils.GetE2EImage(imageutils.Agnhost),
-				Command:      []string{"/agnhost", "porter"},
-				Name:         service.Name,
-				Namespace:    f.Namespace.Name,
-				Replicas:     1,
-				PollInterval: time.Second,
-				Env: map[string]string{
-					"SERVE_PORT_80":   `<a href="/rewriteme">test</a>`,
-					"SERVE_PORT_1080": `<a href="/rewriteme">test</a>`,
-					"SERVE_PORT_160":  "foo",
-					"SERVE_PORT_162":  "bar",
-
-					"SERVE_TLS_PORT_443": `<a href="/tlsrewriteme">test</a>`,
-					"SERVE_TLS_PORT_460": `tls baz`,
-					"SERVE_TLS_PORT_462": `tls qux`,
-				},
-				Ports: map[string]int{
-					"dest1": 160,
-					"dest2": 162,
 
-					"tlsdest1": 460,
-					"tlsdest2": 462,
+			deploymentConfig := e2edeployment.NewDeployment(service.Name,
+				1,
+				labels,
+				service.Name,
+				imageutils.GetE2EImage(imageutils.Agnhost),
+				appsv1.RecreateDeploymentStrategyType)
+			deploymentConfig.Spec.Template.Spec.Containers[0].Command = []string{"/agnhost", "porter"}
+			deploymentConfig.Spec.Template.Spec.Containers[0].Env = []v1.EnvVar{
+				{
+					Name:  "SERVE_PORT_80",
+					Value: `<a href="/rewriteme">test</a>`,
 				},
-				ReadinessProbe: &v1.Probe{
-					ProbeHandler: v1.ProbeHandler{
-						HTTPGet: &v1.HTTPGetAction{
-							Port: intstr.FromInt32(80),
-						},
+				{
+					Name:  "SERVE_PORT_1080",
+					Value: `<a href="/rewriteme">test</a>`,
+				},
+				{
+					Name:  "SERVE_PORT_160",
+					Value: "foo",
+				},
+				{
+					Name:  "SERVE_PORT_162",
+					Value: "bar",
+				},
+				{
+					Name:  "SERVE_TLS_PORT_443",
+					Value: `<a href="/tlsrewriteme">test</a>`,
+				},
+				{
+					Name:  "SERVE_TLS_PORT_460",
+					Value: "tls baz",
+				},
+				{
+					Name:  "SERVE_TLS_PORT_462",
+					Value: "tls qux",
+				},
+			}
+			deploymentConfig.Spec.Template.Spec.Containers[0].Ports = []v1.ContainerPort{
+				{
+					ContainerPort: 80,
+				},
```

Comment:
this was not present in the previous test, should it matter?

Reply:
it was another of those "internal framework magic" cases, https://github.com/kubernetes/kubernetes/blob/2a08ce5c68a3ad47daa0c3be597adc810b3730a5/test/utils/runners.go#L465

and yes, weirdly enough it was making the test fail [here](https://github.com/kubernetes/kubernetes/blob/95860cff1c418ea6f5494e4a6168e7acd1c390ec/test/e2e/network/proxy.go#L204), when accessing the pod directly without specifying port

Reply:
These things are important for the reviewers, my advice is for future PRs you leave a comment here explaining it to save time to the reviewer to figure out the diff